sqlglot/dialects/dialect.py

```diff
@@ -342,8 +342,14 @@ class Dialect(metaclass=_Dialect):
     True means `a / b` is integer division if both `a` and `b` are integers.
     """
 
-    SAFE_DIVISION = False
-    """Whether division by zero throws an error (`False`) or returns NULL (`True`)."""
+    SAFE_DIVISION: bool | str = False
+    """Whether division by zero throws an error (`False`), returns NULL (`True`), or "infinity" (`"inf"`)."""
```

Comment:
I'm considering creating an enumeration for this, similar to `NormalizationStrategy`. Any ideas for naming? I thought of `DivByZeroStrategy`.

Reply:
Chatted with Toby and concluded that we'd rather avoid the enum approach for now.